forecasting/src/autogluon/forecasting/predictor/predictor.py

```diff
@@ -0,0 +1,691 @@
+            validation data used for hyperparameter tuning.
+            If provided, it should have the same format as train_data.
+        hyperparameter_tune_kwargs: Optional, None by default, can be str or dict
+            Valid str values:
+                'auto': Uses the 'bayesopt' preset.
+                'random': Performs HPO via random search using local scheduler.
+                'bayesopt': Performs HPO via bayesian optimization using local scheduler.
+            For valid dictionary keys, refer to :class:`autogluon.core.scheduler.FIFOScheduler` documentation.
+                The 'searcher' key is required when providing a dict.
+                The following is an example dictianary for hyperparameter_tune_kwargs:
+                hyperparameter_tune_kwargs={
+                                            'scheduler': 'local',
+                                            'searcher': 'random',
+                                            'num_trials': 2
+                                            }
+        hyperparameters: Optional, None by default, can be dict
+            parameters for each model. If not provided, default parameters from gluonts will be used.
+
+            Keys are strings that indicate which model types to train.
+            Stable model options include:
+               'DeepAR',
+               'MQCNN',
+               'SFF'(SimpleFeedForward),
+            Experimental model options include:
+               'AutoTabular'(Use Autogluon's TabularPredictor for forecasting)
+            If a certain key is missing from hyperparameters, then `fit()` will not train any models of that type.
+            For example, set `hyperparameters = { 'SFF':{...} }` means you only want to train SimpleFeedForward models and not any other model.
+
+            Values are dictionaries of hyperparameter settings for each model type.
+            Each hyperparameter can either be a single fixed value or a search space containing many possible values.
+            Search Space should only be provided when hyperparameter_tune_kwargs is provided(hyperparameter_tune is turned on)
+
+        time_limits: int, default=None
+            Only works when hyperparameter_tune_kwarg is not None, approximately how long each model will be tunned for.(wallclock time in seconds)
+
+        static_features: pd.Dataframe, default=None
+            static features used for training.
+
+        refit_full: bool, default=False
+            Whether to retrain all models on all of the data (training + validation) after the normal training procedure.
+            If `refit_full=True`, it will be treated as `refit_full='all'`.
+            If `refit_full=False`, refitting will not occur.
+            Valid str values:
+                `all`: refits all models.
+                `best`: refits only the best model (and its ancestors if it is a stacker model).
+
+        set_best_to_refit_full: bool, default=False
```

Comment:
Why not make this = True by default?  Don't we want most novice ML users to automatically benefit from this?

Reply:
This is now changed to:
it is by default False, when refit is set, it is going to be turned to True by default, but the user can still set their own value to override it.